Main.java

```diff
@@ -1,588 +1,287 @@
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.io.Serializable;
+import java.io.*;
 import java.util.ArrayList;
 import java.util.Scanner;
 
-class Food implements Serializable
-{
-    int itemno;
+class Food implements Serializable {
+    int itemNo;
     int quantity;   
     float price;
     
-    Food(int itemno,int quantity)
-    {
-        this.itemno=itemno;
-        this.quantity=quantity;
-        switch(itemno)
-        {
-            case 1:price=quantity*50;
-                break;
-            case 2:price=quantity*60;
-                break;
-            case 3:price=quantity*70;
-                break;
-            case 4:price=quantity*30;
-                break;
+    Food(int itemNo, int quantity) {
+        this.itemNo = itemNo;
+        this.quantity = quantity;
+        this.price = calculatePrice(itemNo, quantity);
+    }
+
+    private float calculatePrice(int itemNo, int quantity) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/shouryaj98/Hotel-Management-Project-Java/pull/54#issuecomment-2437632510>#b99405</a><span> - 10/25/2024, 12:22 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The calculatePrice method lacks input validation for itemNo and quantity
</div>
<div id="fix">
<b>Fix:</b> Add input validation to ensure itemNo is within valid range and quantity is positive
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -16,9 +16,13 @@ class Food implements Serializable
      private float calculatePrice(int itemNo, int quantity) {
 +        if (itemNo < 1 || itemNo > 4) {
 +            throw new IllegalArgumentException("Invalid item number");
 +        }
 +        if (quantity <= 0) {
 +            throw new IllegalArgumentException("Quantity must be positive");
 +        }
          switch (itemNo) {
              case 1: return quantity * 50;
              case 2: return quantity * 60;
              case 3: return quantity * 70;
              case 4: return quantity * 30;
 -            default: return 0;
          }
      }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged